services/command-router/src/main/java/org/eclipse/hono/commandrouter/impl/amqp/ProtonBasedCommandConsumerFactoryImpl.java

```diff
@@ -157,13 +156,17 @@ protected void onDisconnect() {
                 }).map(receiver -> {
                     log.debug("successfully created MappingAndDelegatingCommandConsumer [{}]", address);
                     consumerLinkTenants.add(tenantId);
-                    final CommandConsumer consumer = new CommandConsumer(connection, receiver);
-                    consumer.setLocalCloseHandler(sourceAddress -> {
-                        log.debug("MappingAndDelegatingCommandConsumer receiver link [tenant-id: {}] closed locally", tenantId);
-                        mappingAndDelegatingCommandConsumerFactory.removeClient(tenantId);
-                        consumerLinkTenants.remove(tenantId);
-                    });
-                    return (MessageConsumer) consumer;
+                    return (CommandConsumer) new CommandConsumer() {
```

Comment:
do we really need the cast here?

Reply:
It seems to be needed, yes. Without it there is this compile error:
````
incompatible types: 
io.vertx.core.Future<<anonymous org.eclipse.hono.adapter.client.command.CommandConsumer>> 
cannot be converted to 
io.vertx.core.Future<org.eclipse.hono.adapter.client.command.CommandConsumer>
````

Reply:
interesting